docs/tutorials/10_effective_dimension.ipynb

```diff
@@ -0,0 +1,763 @@
+      "text/plain": [
+       "<Figure size 447.797x204.68 with 1 Axes>"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "num_qubits = 3\n",
+    "# create a feature map\n",
+    "feature_map = ZFeatureMap(feature_dimension=num_qubits, reps=1)\n",
+    "# create a variational circuit\n",
+    "ansatz = RealAmplitudes(num_qubits, reps=1)\n",
+    "\n",
+    "# create quantum circuit\n",
+    "qc = QuantumCircuit(num_qubits)\n",
+    "qc.append(feature_map, range(num_qubits))\n",
+    "qc.append(ansatz, range(num_qubits))\n",
+    "qc.decompose().draw(\"mpl\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The parametrized circuit can then be sent together with an optional interpret map (parity in this case) to the `CircuitQNN` constructor."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# parity maps bitstrings to 0 or 1\n",
+    "def parity(x):\n",
+    "    return \"{:b}\".format(x).count(\"1\") % 2\n",
+    "\n",
+    "\n",
+    "output_shape = 2  # corresponds to the number of classes, possible outcomes of the (parity) mapping."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# declare quantum instance\n",
+    "qi_sv = QuantumInstance(Aer.get_backend(\"aer_simulator_statevector\"))\n",
+    "\n",
+    "# construct QNN\n",
+    "qnn = CircuitQNN(\n",
+    "    qc,\n",
+    "    input_params=feature_map.parameters,\n",
+    "    weight_params=ansatz.parameters,\n",
+    "    interpret=parity,\n",
+    "    output_shape=output_shape,\n",
+    "    sparse=False,\n",
+    "    quantum_instance=qi_sv,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3.1 Define the Problem\n",
+    "\n",
+    "In order to define the problem, we need a series of sets of input samples (`num_input_samples`) and weights (`num_weight_samples`), as well as the total number of data samples available in a dataset (`n`). The `input_samples` and `weight_samples` are set in the class constructor, while the number of data samples is given during the call to the effective dimension computation to be able to test and compare how this measure changes with different dataset sizes.\n",
```

Comment:
> (`num_input_samples`) and weights (`num_weight_samples`)

Here there are what look like variables/properties e..g `num_input_samples` that might be accessed. These don't exist though right its just the number of input samples ie the first dimension of the input samples array etc.

Reply:
True, they were accessible at the beginning but not any more. Will change.